### ISSUE TITLE
Support for content hash in output filename

### ISSUE THREAD

**Issue by CreepGin** [fuse-box/fuse-box#104]
Webpack supports special keywords in the output filename. 

```
output: {
    path: "./public/build",
    filename: PRODUCTION ? "[name]___[chunkhash].script.min.js" : "[name].script.js"
},
```

Among other things, this is most often used for asset cache busting. However the webpack way may be a bit too free-form. So in FuseBox, why not just assume this behavior upon a boolean flag or when `NODE_ENV === "production"`.

```
filenameHash: true
```

or just with

```
fsbx.EnvPlugin({ NODE_ENV: “production" })
```

**Comment by nchanged** [comment]
You also need to understand that FuseBox does not have chunks. We have `fuses` if we can call them so. You can create as many bundles as you like and they will fuse together at runtime.

```
outFile : "./bundle-[hash].js"
```

**Comment by unlight** [comment]
My requirements to hashify
1. Should not depends on other plugins (env plugin, etc.)
2. Allow to specify custom value (I want include version number)
3. Do hashify for ALL output files including js, css, images
4. Dynamic import should respect hash option

**Comment by nchanged** [comment]
@unlight what is it about `Dynamic import should respect hash option` 
can you give me an example?

**Comment by unlight** [comment]
Example
```
fuse.bundle({
'build/app.js': '>main.ts'
build/about.js: >about.ts
})
```
In app.js `FuseBox.import(about.js)` must load `about[hash].js`

**Comment by CreepGin** [comment]
@nchanged For the comment about `chunkhash`, webpack generally has 3 keywords for hashes: `hash`, `chunkhash`, and `contenthash` (the last one is only supported by `ExtractTextPlugin`).

I used `chunkhash` in the example because it was the most reliable (for webpack). But generally speaking for FuseBox, a content hash of the actual resulting file will do the job. So in my understanding, you don't really need to worry about chunks and fuses.

**Comment by nchanged** [comment]
@CreepGin and, I presume, it's `md5` sum, correct?

**Comment by CreepGin** [comment]
Also to specify a custom value (per unlight's suggestion), what do you think about the following:

```
outFile: function(meta) {
    return meta.filename.replace(".js", `.${meta.hash}.js`);
}
```

You can do whatever you want in the callback (appending the hash or your custom version number as needed). `meta` contains meta information about the file.

**Comment by CreepGin** [comment]
@nchanged correct

**Comment by leon-andria** [comment]
hi guys,
```
outFile: function(meta) {
    return meta.filename.replace(".js", `.${meta.hash}.js`);
}
```
This syntax will pollute the clean syntax config that we have now, and we should stick this in mind as much as possible.

If we want to manage a kind of replacement parameters, we should be inspired by MSBuild. Fuse config can take benefit of parameters coming from outside during initialization.

**Comment by devmondo** [comment]
@leon-andria MSBuild!!! god, Fuse makes it look like Dinosaur, but all valid points, and this is what makes Fuse better, so rest assured we are working to keep it always this way :)

**Comment by leon-andria** [comment]
Good that's you like the idea. So have this syntax will be natural,
```
outFile: '${filename}-${hash}.min.js'
```

The idea is that any string in fuse config can be like a string interpolate :-)

**Comment by nchanged** [comment]
https://github.com/fuse-box/fuse-box/issues/350

**Comment by nchanged** [comment]
Implemented in 1.4.1
